src/status.c

```diff
@@ -0,0 +1,104 @@
+#include <stdio.h> // Must be included before ebmusv2.h
+
+#include "ebmusv2.h"
+#include <assert.h>
+#include <commctrl.h>
+#include <stdarg.h>
+
+void format_status(int part, const char* format, ...) {
+	if (hwndStatus) {
+		va_list args;
+		va_start(args, format);
+		int size = vsnprintf(0, 0, format, args);
+		va_end(args);
+
+		char *buf = malloc(size + 1);
+		va_start(args, format);
+		vsprintf(buf, format, args);
+		va_end(args);
+		
+		SendMessage(hwndStatus, SB_SETTEXT, part, (LPARAM)buf);
+
+		free(buf);
+	}
+}
+
+static const char* notes[] = { "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B" };
+
+void set_tracker_status(int p, BYTE *code) {
+	if (code[0] == 0x00) {
+		format_status(p, "End of pattern/subroutine");
+	} else if (code[0] < 0x80) {
+		if (code[1] < 0x80) {
+			format_status(p, "Note length: %d, velocity: %d/7, release: %d/15", code[0], (code[1] >> 4) & 0x7, code[1] & 0xF);
+		} else {
+			format_status(p, "Note length: %d", code[0]);
+		}
+	} else if (code[0] < 0xC8) {
+		BYTE note = (code[0] & 0x7F);
+		format_status(p, "Note %s%c", notes[note % 12], '1' + note / 12);
+	} else if (code[0] == 0xC8) {
+		format_status(p, "Hold");
+	} else if (code[0] == 0xC9) {
+		format_status(p, "Rest");
+	} else if (code[0] < 0xE0) {
+		format_status(p, "Percussion %d", code[0] - 0xC9);
+	} else {
+		switch (code[0]) {
+			case 0xE0: format_status(p, "Instrument %d", code[1]); break;
+			case 0xE1: format_status(p, "Pan %d", code[1]); break;
+			case 0xE2: format_status(p, "Pan (duration: %d, pan: %d)", code[1], code[2]); break;
+			case 0xE3:
+				format_status(p, "Enable vibrato (delay: %d, freq: %d, range: \xB1%d%s)",
+					code[1],
+					code[2],
+					code[3] <= 0xF0 ? code[3] : code[3] - 0xF0,
+					code[3] <= 0xF0 ? "/256 semitones" : " semitones");
+				break;
+			case 0xE4: format_status(p, "Disable vibrato"); break;
+			case 0xE5: format_status(p, "Master volume %d/255", code[1]); break;
+			case 0xE6: format_status(p, "Master volume (duration: %d, volume: %d/255)", code[1], code[2]); break;
+			case 0xE7: format_status(p, "Tempo %d", code[1]); break;
+			case 0xE8: format_status(p, "Tempo (duration: %d, tempo: %d)", code[1], code[2]); break;
+			case 0xE9: format_status(p, "Global transpose %d", (signed char)code[1]); break;
+			case 0xEA: format_status(p, "Channel transpose %d", (signed char)code[1]); break;
```

Comment:
Personal preference: `%d` --> `%+d`, so that it shows as +1 for positive numbers. I can confirm this works correctly with msvcrt.

Reply:
Done